deployment/src/main/java/io/quarkiverse/poi/deployment/POIProcessor.java

```diff
@@ -58,4 +58,13 @@ void registerXMLBeansClassesForReflection(CombinedIndexBuildItem combinedIndexBu
             reflectiveClass.produce(new ReflectiveClassBuildItem(false, true, implementor.name().toString()));
         }
     }
+
+    @BuildStep
+    public ReflectiveClassBuildItem registerLog4jClassesForReflection() {
+        return new ReflectiveClassBuildItem(true, true,
+                "org.apache.logging.log4j.message.ReusableMessageFactory",
+                "org.apache.logging.log4j.message.DefaultFlowMessageFactory",
+                "org.apache.logging.log4j.message.ParameterizedMessageFactory");
+    }
```

Comment:
I thought I posted a comment here but apparently not. I’m surprised you have to register theses classes as you shouldn’t use the impl just the API and the JBoss Logging adapter. 

Is it somehow required by the API? Do we need to adjust the adapter?

(asking because we have a lot of other extensions using Log4j 2)

Reply:
You did here: https://github.com/quarkiverse/quarkus-poi/pull/30#discussion_r1118781191

As I said classes are part of the Log4j 2 API JAR. I think it would be useful if we added that in the adapter.